pkg/utils/utils.go

```diff
@@ -83,7 +84,19 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 	} else if clusterFlavor == constants.Supervisor {
 		ns = constants.VCSecretNsSupervisor
 	} else {
-		ns = constants.VCSecretNs
+		// check the current CSI version for vanilla setups.
+		// If >=2.3, then the secret is in vmware-system-csi
+		// If <2.3, then the secret is in kube-system
```

Comment:
So we are still relying on vSphere CSI driver's secrets?  I thought we are going to get rid of this dependency?
Note that there are cases where a vSphere CSI Driver deployment relies on a config file, not secrets:
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/manifests/vanilla/vsphere-csi-driver.yaml#L189-L190

Reply:
Yes, we will still rely on csi driver secrets, to remove dependency we need code changes from WCP side, so as of now we will continue with the current behavior.
The config file is actually the secret mounted into the csi driver as a volume(https://vsphere-csi-driver.sigs.k8s.io/driver-deployment/installation.html)

Reply:
I verified with Divyen that a config file without secret is supported by vSphere CSI Driver.  TKGi deploys vSphere CSI Driver as process not as pod.